core/server/i18n.js

```diff
@@ -127,9 +181,79 @@ I18n = {
+            });
+
+            if (!hasBuiltInLocaleData) {
+                // `Intl` exists, but it doesn't have the data we need, so load the
+                // polyfill and replace the constructors with need with the polyfill's.
+                IntlPolyfill = require('intl');
+                Intl.NumberFormat   = IntlPolyfill.NumberFormat;
+                Intl.DateTimeFormat = IntlPolyfill.DateTimeFormat;
+            }
+        } else {
+            // No `Intl`, so use and load the polyfill.
+            global.Intl = require('intl');
+        }
+    },
+
+    /**
+     * Setup i18n support for themes:
+     *  - Load proper language file in to memory
+     *  - Polyfill node.js if it does not have Intl support or support for a particular locale
+     */
+    loadThemeTranslations: function loadThemeTranslations() {
+        // This function is called during theme initialization, and when switching language or theme.
+
+        var blosTheme, hasBuiltInLocaleData, IntlPolyfill;
+
+        currentLocale = I18n.locale();
+        activeTheme = settingsCache.get('active_theme') || '';
+
+        // Reading file for current locale and active theme and keeping its content in memory
+        if (activeTheme) {
+            // Reading translation file for theme .hbs templates.
+            // Compatibility with both old themes and i18n-capable themes.
+            // Preventing missing files.
+            try {
+                blosTheme = fs.readFileSync(path.join(__dirname, '..', '..', 'content', 'themes', activeTheme, 'locales', currentLocale + '.json'));
+            } catch (err) {
+                blosTheme = undefined;
+                if (err.code === 'ENOENT') {
+                    logging.warn('Theme\'s file locales/' + currentLocale + '.json not found.');
+                } else {
+                    throw err;
+                }
+            }
+            if (blosTheme === undefined && currentLocale !== 'en') {
+                logging.warn('Falling back to locales/en.json.');
+                try {
+                    blosTheme = fs.readFileSync(path.join(__dirname, '..', '..', 'content', 'themes', activeTheme, 'locales', 'en.json'));
+                } catch (err) {
+                    blosTheme = undefined;
+                    if (err.code === 'ENOENT') {
+                        logging.warn('Theme\'s file locales/en.json not found.');
+                    } else {
+                        throw err;
+                    }
+                }
+            }
+            if (blosTheme !== undefined) {
+                // if translation file is not valid, you will see an error
+                try {
+                    blosTheme = JSON.parse(blosTheme);
+                } catch (err) {
+                    blosTheme = undefined;
+                    throw err;
+                }
+                blos = _.merge(blos, blosTheme);
+            }
+        }
 
+        if (global.Intl) {
```

Comment:
I think we can remove this whole part, because we dropped Node 0.10?
See https://github.com/yahoo/intl-messageformat#loading-intljs-polyfill-in-nodejs.

Reply:
Maybe yes, maybe no.

On the one hand, [Intl.js](https://github.com/andyearnshaw/Intl.js) people say _"Node.js versions prior to 0.12 and ≥v3.1 don't provide the Intl APIs, so they require that the runtime is polyfilled."_ Probably it's more accurate what the Node.js [docs](https://nodejs.org/api/intl.html) (and [old version](https://github.com/nodejs/node/wiki/Intl)) seem to suggest: by default (that is, `small-icu` option) the Intl API is included, but only with the English locale data. Another possibility is to build Node.js with all locales, using `./configure --with-intl=full-icu --download=all` (optional due to size, etc.).

On the other hand, I think Ghost is not really using Intl right now, but a different i18n system, plural helper, moment.js, etc. However Intl is now a standard and could be used for future localization features (NumberFormat?).

So you can say me if we keep the polyfill just in case, or we remove it for now and add it again later when needed. I think maybe the second option, if it's not needed now or soon.

Reply:
Reviewing the polyfill more carefully, sorry, it looks like Ghost is indeed using it for `intl-messageformat` in `18n.t()`, since by default Node.js includes the `Intl` API but only with the English locale data. So, better we keep the polyfill for now.